src/page/LogInAndOut/Signup.jsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import { TextField, Button, Container, Typography } from "@mui/material";
+import { useNavigate } from "react-router-dom";
+import bcrypt from "bcryptjs";
+
+const generateToken = () => {
+    return Math.random().toString(36).substr(2) + Date.now().toString(36);
+  };
+const Signup = () => {
+    const navigate = useNavigate();
+
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+
+  const handleSignup = async () => {
+   
+        if (password !== confirmPassword) {
+          alert("كلمة المرور وتأكيد كلمة المرور غير متطابقتين.");
+          return;
+        }
+    const users = JSON.parse(localStorage.getItem("users")) || [];
+     const userExists = users.some(user => user.username === username);
+     if (userExists) {
+       alert("اسم المستخدم موجود مسبقًا.");
+       return;
+     }
+
+     const hashedPassword = await bcrypt.hash(password, 10);
+ 
+    const token = generateToken();
+    const newUser = { username, password: hashedPassword, token };
+    users.push(newUser);
+    localStorage.setItem("users", JSON.stringify(users));
```

Comment:
in our case the local storage used instead of database so the token is not stored in the users table (localstorage array in this case ).
**I don't want you to store the token in the users object and array**